src/sync/sync-utils.ts

```diff
@@ -110,9 +110,8 @@ const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncTyp
 
 };
 
-const resetFailedCode = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]) => {
-	// a full sync without target tasks has reposyncstates removed so dont update.
-	if (syncType === "full" && !targetTasks?.length) {
+const resetFailedCode = async (subscription: Subscription, targetTasks?: TaskType[]) => {
+	if (!targetTasks?.length) {
```

Comment:
Curious if it affects the behaviour of https://hello.atlassian.net/wiki/spaces/OTFS/pages/1824654901/HOW+TO+Run+a+backfill+resync+endpoint ?

Reply:
This sync-utils is too complicated, too many branches. 
From what I can tell, this change should only affecting `sync=partial && targetTask == empty`. Current behaviour for that branch is do reset the failCode. And I think the current behaviour is wrong. The fix will make it NOT reset the failCode. So regardless of the runbooks, this fix should works correctly.

--
I can double check the runbook, but doubt any runbook relies on the "wrong" behaviour (that reset failCode for partial and empty target task).

Reply:
Yes, that's why I"m asking :) IMHO we should update our runbook with the change, because it is super compliated and must be documented... wdyt?

Reply:
Will update the doc.